### DISCUSSION TITLE
Strategies to reduce jitter / stutter

### DISCUSSION THREAD

**Discussion by benny-noumena** [jacomyal/sigma.js#1445] [Q&A]
Hey there!

I ran into the issue where with many nodes (~1k) the force layout (used in a worker) produces fatal jitter as the forces may drag the node towards the opposite directions. This can even lead to a severe effect on people with epileptic issues and I would like to avoid this somehow.

The node positions are calculated by `graphology` but it seems that there is no protection for this. I thought I can handle it by measure the delta-distance and delta-angle in the `nodeReducer` but unfortunately it looks like changing `x` and `y` attribute at this point does not have any effect.

What would be the best strategy to address this problem?

Many thanks!
Benny

https://github.com/user-attachments/assets/51feb396-1def-4753-bb9d-f60900d5d2ea

CC: @Yomguithereal

**Comment by sim51** [comment]
The force layout works well form small graphs. But for bigger graphs, I recommend you to use force atlas

**Reply by benny-noumena** [reply]
Sadly it is `force-altas` :(

**Comment by jacomyal** [comment]
This is a very difficult topic. The best I can tell you is to play with the FA2 settings, to find how to use them to reduce jitter **for you specific graphs**.

One tip though: You can try to use [Gephi Lite](https://gephi.org/gephi-lite/) to fine tune the FA2 settings with a UI. The good news is that it uses the same implementation, and the form matches the available function settings.

![image](https://github.com/user-attachments/assets/1abd6f55-3a06-4a82-856d-166ec848122e)
